internal/connectors/engine/engine.go

```diff
@@ -304,10 +304,12 @@ func (e *engine) CreateFormancePayment(ctx context.Context, payment models.Payme
 }
 
 func (e *engine) ForwardBankAccount(ctx context.Context, bankAccountID uuid.UUID, connectorID models.ConnectorID, waitResult bool) (models.Task, error) {
+	id := fmt.Sprintf("create-bank-account-%s-%s", connectorID.String(), bankAccountID.String())
```

Comment:
I still think it would be best to have such ID creation functions in the `models` package to ensure that any package needing to be able to interact with tasks is able to generate the ID in a uniform way like `id := models.Reference(connectorID, bankAccountID)`. Otherwise we end up with little sprintf functions littered around the code base, which may or may not be consistent.

From a division of responsibility perspective, the engine package shouldn't be responsible for how to format reference IDs, if that makes sense?

Reply:
Something like that you mean ? https://github.com/formancehq/payments/pull/142/commits/c21b0e62c3613c0769b0b29c0ca17dd262db280a

Reply:
yes exactly! ✨